pkg/dashboards/dashboards.go

```diff
@@ -579,228 +524,96 @@ func ProcessFavoriteRequest(ctx *fasthttp.RequestCtx) {
-		utils.SetBadMsg(ctx, "")
-		return
-	}
-	utils.WriteJsonResponse(ctx, dIds)
-	ctx.SetStatusCode(fasthttp.StatusOK)
-}
-
-func checkAndReturnFieldInMapIfExists(mapData map[string]interface{}, fieldName string) (interface{}, error) {
-	value, exists := mapData[fieldName]
-	if !exists {
-		return nil, errors.New(fieldName + " field not found")
-	}
-	return value, nil
-}
-
-func parseUpdateDashboardRequest(readJSON map[string]interface{}) (string, string, map[string]interface{}, error) {
-
-	value, err := checkAndReturnFieldInMapIfExists(readJSON, "id")
-	if err != nil {
-		return "", "", nil, err
-	}
-
-	dId, ok := value.(string)
-	if !ok {
-		return "", "", nil, errors.New("id field is not a string")
-	}
-
-	value, err = checkAndReturnFieldInMapIfExists(readJSON, "name")
-	if err != nil {
-		return "", "", nil, err
-	}
-	dName, ok := value.(string)
-	if !ok {
-		return "", "", nil, errors.New("name field is not a string")
-	}
-
-	value, err = checkAndReturnFieldInMapIfExists(readJSON, "details")
-	if err != nil {
-		return "", "", nil, err
-	}
-	dashboardDetails, ok := value.(map[string]interface{})
-	if !ok {
-		return "", "", nil, errors.New("details field is not a map")
-	}
-	return dId, dName, dashboardDetails, nil
-}
-
 func ProcessUpdateDashboardRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 	rawJSON := ctx.PostBody()
 	if rawJSON == nil {
-		log.Errorf("ProcessCreateDashboardRequest: received empty user query")
+		log.Errorf("ProcessUpdateDashboardRequest: received empty request body")
 		utils.SetBadMsg(ctx, "")
 		return
 	}
 
 	readJSON := make(map[string]interface{})
-
-	err := json.Unmarshal(rawJSON, &readJSON)
-	if err != nil {
-		log.Errorf("ProcessCreateDashboardRequest: could not unmarshall body: %v, err=%v", rawJSON, err)
-		utils.SetBadMsg(ctx, "")
+	if err := json.Unmarshal(rawJSON, &readJSON); err != nil {
+		log.Errorf("ProcessUpdateDashboardRequest: could not unmarshal body: %v, err=%v", rawJSON, err)
+		utils.SetBadMsg(ctx, "Invalid request body")
 		return
 	}
 
 	dId, dName, dashboardDetails, err := parseUpdateDashboardRequest(readJSON)
 	if err != nil {
-		log.Errorf("ProcessCreateDashboardRequest: parseUpdateDashboardRequest failed, readJSON: %v, err: %v", readJSON, err)
-		utils.SetBadMsg(ctx, "")
+		log.Errorf("ProcessUpdateDashboardRequest: parseUpdateDashboardRequest failed, readJSON: %v, err: %v", readJSON, err)
+		utils.SetBadMsg(ctx, "Invalid request format")
 		return
 	}
+
+	// Panel validation if needed
 	if panels, ok := dashboardDetails["panels"].([]interface{}); ok {
 		if pflag, ok := dashboardDetails["panelFlag"]; ok {
 			if pflag == "false" && len(panels) > 10 {
+				utils.SetBadMsg(ctx, "Too many panels for free tier")
 				return
 			}
 		}
 	}
+
+	// Update the dashboard
 	err = updateDashboard(dId, dName, dashboardDetails, myid)
 	if err != nil {
-		if err.Error() == "dashboard name already exists" {
+		switch err.Error() {
+		case "dashboard name already exists in this folder":
```

Comment:
This looks like something you're returning from javascript probably? I'd recommend adding a comment to make that clear (e.g., `// keep this synced with dashboardClashMsg in foo.js`). Also, if you're using this exact string more than once in the backend, I'd recommend making a constant for it